src/irmin/tree.ml

```diff
@@ -1760,8 +1760,12 @@ module Make (P : Private.S) = struct
 
   module Proof = struct
     type tree = t
+
+    include Proof.Make (P.Hash) (Path) (Metadata)
```

Comment:
`Proof.Make.t` and `Proof.Make.proof` are different. This is a bit hard to follow 😁

Reply:
`t` is `(before_hash * proof * after_hash)`. I can make this clearer.

Reply:
I've added some docs in e695312

Reply:
Thanks a lot. I'll continue reviewing